dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/advanced/DateRangeValidator.java

```diff
@@ -0,0 +1,106 @@
+package edu.harvard.iq.dataverse.search.advanced;
+
+import edu.harvard.iq.dataverse.common.BundleUtil;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.time.DateUtils;
+
+import javax.faces.application.FacesMessage;
+import javax.faces.component.UIComponent;
+import javax.faces.component.UIInput;
+import javax.faces.context.FacesContext;
+import javax.faces.validator.FacesValidator;
+import javax.faces.validator.Validator;
+import javax.faces.validator.ValidatorException;
+import java.text.ParseException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+
+@FacesValidator(value = "dateRangeValidator")
+public class DateRangeValidator implements Validator {
+
+    // -------------------- LOGIC --------------------
+    @Override
+    public void validate(FacesContext context, UIComponent component, Object value) throws ValidatorException {
+        UIInput dateFromInput = (UIInput) component.getAttributes().get("dateFrom");
+        UIInput dateToInput = (UIInput) component.getAttributes().get("dateTo");
+
+        Object dateFromValue = dateFromInput.getSubmittedValue();
+        Object dateToValue = dateToInput.getSubmittedValue();
+
+        LocalDate dateFrom = validateDateField(context, dateFromInput, dateFromValue);
+        LocalDate dateTo = validateDateField(context, dateToInput, dateToValue);
+
+        boolean isFromAfterTo = false;
+        if(dateFrom != null && dateTo != null) {
+            isFromAfterTo = dateFrom.isAfter(movePartialDateToUpperLimit(dateTo, dateToValue.toString()));
+        }
+
+        if(isFromAfterTo) {
+            dateFromInput.setValid(false);
+            dateToInput.setValid(false);
+            FacesMessage message = new FacesMessage(FacesMessage.SEVERITY_ERROR, "", BundleUtil.getStringFromBundle("advanced.search.wrong.daterange.badRange"));
+            context.addMessage(dateFromInput.getClientId(context), message);
+            context.addMessage(dateToInput.getClientId(context), message);
+        }
+    }
+
+    // -------------------- PRIVATE ---------------------
+    private LocalDate validateDateField(FacesContext context, UIComponent comp, Object value) {
```

Comment:
There are too many different types of logic here in this method: validation, view and parsing.

Reply:
What's your suggestion then?

Reply:
1. One method only for validation
2. Another method for display
3. The result of the first is passed to second in main method of the class.